src/components/common/form/message-item/MessageItem.jsx

```diff
@@ -0,0 +1,38 @@
+import PropTypes from 'prop-types';
+import SingleDateMoved from '../message-item-types/SingleDateMoved';
+import DoubleDateMoved from '../message-item-types/DoubleDateMoved';
+
+const components = {
+  singleDateMoved: SingleDateMoved,
+  doubleDateMoved: DoubleDateMoved,
+};
+
+const MessageItem = ({
+  messageSummary,
+  setSummaryMessages,
+  setIsAlertVisible,
+}) => {
+  if ('update' in messageSummary) {
+    const template = messageSummary['update'].template;
+    const variables = messageSummary['update'].variables;
+
+    const Component = components[template];
+    return (
+      <Component
+        variables={variables}
+        setSummaryMessages={setSummaryMessages}
+        setIsAlertVisible={setIsAlertVisible}
+      />
+    );
```

Comment:
This doesn't feel very React like, and also it results in us having duplication between `SingleDateMoved` and `DoubleDateMoved`.
What we're trying to achieve looks like it could be solved through React composition:
https://reactjs.org/docs/composition-vs-inheritance.html
Based on React's suggestion for a typical inheritance situation it looks like we could have a base like `DateMoved` component which contains whats currently in `SingleDateMoved` and then any additional ui e.g. what's in `DoubleDateMoved` could be  passed in as `children`. 
That way you could remove the need to rely on passing the string name of a component to render it and the duplication across both components.
Happy to discuss, maybe it's not an issue but probs worth while considering.